src/Instrumentation/Laravel/src/Watchers/RedisCommand/RedisCommandWatcher.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\RedisCommand;
+
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Redis\Connections\Connection;
+use Illuminate\Redis\Connections\PhpRedisConnection;
+use Illuminate\Redis\Connections\PredisConnection;
+use Illuminate\Redis\Events\CommandExecuted;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\Contrib\Instrumentation\Laravel\Watchers\Watcher;
+use OpenTelemetry\SemConv\TraceAttributes;
+use OpenTelemetry\SemConv\TraceAttributeValues;
+use RangeException;
+use RuntimeException;
+use Throwable;
+
+/**
+ * Watch the Redis Command event
+ *
+ * Call facade `Redis::enableEvents()` before using this watcher
+ */
+class RedisCommandWatcher extends Watcher
+{
+    public function __construct(
+        private CachedInstrumentation $instrumentation,
+    ) {
+    }
+
+    /** @psalm-suppress UndefinedInterfaceMethod */
+    public function register(Application $app): void
+    {
+        /** @phan-suppress-next-line PhanTypeArraySuspicious */
+        $app['events']->listen(CommandExecuted::class, [$this, 'recordRedisCommand']);
+    }
+
+    /**
+     * Record a query.
+     */
+    /** @psalm-suppress UndefinedThisPropertyFetch */
+    public function recordRedisCommand(CommandExecuted $event): void
+    {
+        $nowInNs = (int) (microtime(true) * 1E9);
+
+        $operationName = strtoupper($event->command);
+
+        /** @psalm-suppress ArgumentTypeCoercion */
+        $span = $this->instrumentation->tracer()
+            ->spanBuilder($operationName)
+            ->setSpanKind(SpanKind::KIND_CLIENT)
+            ->setStartTimestamp($this->calculateQueryStartTime($nowInNs, $event->time))
+            ->startSpan();
+
+        // See https://opentelemetry.io/docs/specs/semconv/database/redis/
+        $attributes = [
+            TraceAttributes::DB_SYSTEM => TraceAttributeValues::DB_SYSTEM_REDIS,
+            TraceAttributes::DB_NAME => $this->fetchDbIndex($event->connection),
+            TraceAttributes::DB_OPERATION => $operationName,
+            TraceAttributes::DB_QUERY_TEXT => Serializer::serializeCommand($event->command, $event->parameters),
```

Comment:
This appears to be unavailable in semconv 1.24, which is the target for a 1.0 release (and the last supported version for php 8.0).

Not sure if there is an alternative which might be better suited here?

Reply:
Use DB_STATEMENT for compatibility with semconv 1.24.

Ref: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.19.0/specification/trace/semantic_conventions/database.md#redis